### PR TITLE
extract ui to console-ui

### PR DESCRIPTION
**Pull request by stefanpenner**
The idea is console-ui to describe a common API for other libraries to confirm to. So that, there ui output can work nicely within a larger system.

* ui implementation
* mock ui implementation

This enables blueprint extraction

### CODE REVIEW

**Comment by stefanpenner** [comment]
code coverage decreased, but only because these files where bumping up the overall ratio. Not that test coverage actually went down, as these tests still run just in there own repository

**Comment by rwjblue** [comment]
@homu r+

File: lib/models/project.js [hunk 1]
```diff
@@ -686,7 +686,7 @@ function ensureUI(_ui) {
 
   if (!ui) {
     // TODO: one UI (lib/cli/index.js also has one for now...)
-    var UI = require('../ui');
+    var UI = require('command-ui');
```
Comment (Turbo87) [inline lib/models/project.js]: typo: `console-ui`

File: lib/cli/index.js [hunk 2]
```diff
@@ -58,7 +58,7 @@ function configureLogger(env) {
 
 // Options: Array cliArgs, Stream inputStream, Stream outputStream
 module.exports = function(options) {
-  var UI = options.UI || require('../ui');
+  var UI = options.UI || require('command-ui');
```
Comment (Turbo87) [inline lib/cli/index.js]: typo: `console-ui`

**Pull request merged**